tgui/packages/tgui/interfaces/PipeDispenser.js

```diff
@@ -0,0 +1,137 @@
+/**
+ * Copyright (c) 2023 @Laboredih123
+ * SPDX-License-Identifier: MIT
+ */
+
+import { useBackend, useLocalState } from '../backend';
+import { Button, Section, Flex, Box, NumberInput, Image } from '../components';
+import { Window } from '../layouts';
+
+export const PipeDispenser = (props, context) => {
+  const { data } = useBackend(context);
+  const disposalpipes = data.disposalpipes || [];
+  const {
+    dispenser_ready,
+    windowName,
+    mobile,
+    removing_pipe,
+    laying_pipe,
+    max_disposal_pipes,
+  } = data;
+  return (
+    <Window
+      title={windowName}
+      width="310"
+      height={mobile ? 360 : 270}>
+      <Window.Content scrollable>
+        <Section>
+          {disposalpipes.map(disposalpipe => {
+            return (
+              <DisposalPipeRow
+                key={disposalpipe.disposaltype}
+                dispenser_ready={dispenser_ready}
+                max_disposal_pipes={max_disposal_pipes}
+                disposalpipe={disposalpipe} />
+            );
+          })}
+        </Section>
+        {!!mobile && (
+          <AutoPipeLaying
+            laying_pipe={laying_pipe}
+            removing_pipe={removing_pipe} />
+        )}
+      </Window.Content>
+    </Window>
+  );
+};
+
+export const DisposalPipeRow = (props, context) => {
+  const { act } = useBackend(context);
+  const [amount, setAmount] = useLocalState(context, 'amount', 1);
+  const {
+    dispenser_ready,
+    max_disposal_pipes,
+    disposalpipe,
+  } = props;
+
+  return (
+    <Flex justify="space-between" align="stretch" style={{ "border-bottom": "1px #555 solid" }}>
```

Comment:
I think that a chunk of the props on here are probably unnecessary. Consider using a `Stack` instead of `Flex` to get some of the spacing that you add in otherwise for "free".

Reply:
Looking further into, using a `Stack` here and should let you remove a lot of the `style` attributes on the `.Item`s.

Reply:
think i removed the unnecessary ones